pkgs/tools/filesystems/xfsprogs/default.nix

```diff
@@ -19,7 +20,8 @@ stdenv.mkDerivation rec {
 
   outputs = [ "bin" "dev" "out" "doc" ];
 
-  nativeBuildInputs = [ autoconf automake libtool gettext ];
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
+  nativeBuildInputs = [ autoconf automake libtool gettext libuuid ];
   propagatedBuildInputs = [ utillinux ]; # Dev headers include <uuid/uuid.h>
```

Comment:
Perhaps its like below,

```
 23   depsBuildBuild = [ buildPackages.stdenv.cc ];
 24   nativeBuildInputs = [ autoconf automake libtool gettext libuuid.dev ];
 25   propagatedBuildInputs = [ utillinux ]; # Dev headers include <uuid/uuid.h>                                                                                                                                                               
 26   buildInputs = [ readline libuuid ];
```

How does it look?

Reply:
This seems to be only a build-time dependency: https://bugzilla.redhat.com/show_bug.cgi?id=576296

Reply:
Does utillinux have anything to do with this? Is there any `uuid` in the output? I suspect that comment should be deleted as it has no value. Additional utillinux should probably not be propagated, we should also use @matthewbauer's unix utils to get just the execuables we need.

Reply:
Hey @Ericson2314,

OK, lets run down the questions one by one, I'd like to see how you handle the details.
1. When I try the following: 
```
 24   nativeBuildInputs = [ autoconf automake libtool gettext libuuid ];
 25   #propagatedBuildInputs = [ utillinux ]; # Dev headers include <uuid/uuid.h>                 
```
I get,
```
  9990 checking for sys/uuid.h... no
  9991 checking uuid/uuid.h usability... no
  9992 checking uuid/uuid.h presence... no
  9993 checking for uuid/uuid.h... no
  9994 
  9995 FATAL ERROR: could not find a valid UUID header.
  9996 Install the Universally Unique Identifiers development package.
  9997 builder for '/nix/store/p5wi6m6ysms7j5b6mxq5017hcsx6mydn-xfsprogs-4.14.0-armv7l-unknown-linux-gnueabihf.drv' failed with exit code 1
  9998 error: build of '/nix/store/p5wi6m6ysms7j5b6mxq5017hcsx6mydn-xfsprogs-4.14.0-armv7l-unknown-linux-gnueabihf.drv' failed

```
It seems utillinux provides the headers instead of libuuid. (strange)

2. I don't have any packages that depends on this one to try whether utillinux is really propagated or not so, please let me know.

3. To see any references, I did

```
 23   depsBuildBuild = [ buildPackages.stdenv.cc ];
 24   nativeBuildInputs = [ autoconf automake libtool gettext libuuid ];
 25   propagatedBuildInputs = [ utillinux ]; # Dev headers include <uuid/uuid.h>                                                                                                                                                               
 26   buildInputs = [ readline ];
 27   disallowedReferences = [ autoconf automake libtool gettext libuuid  utillinux ];
```
and it passed. If stdenv does not magically removes them, there are no references in the output.

4. I have no info on that I'm afraid. 

best.

Reply:
Finally, not that the current level of confusion is enough, the one below does not work.

```
 23   depsBuildBuild = [ buildPackages.stdenv.cc ];
 24   nativeBuildInputs = [ autoconf automake libtool gettext ] ; #libuuid ];                                                                                                                                                                  
 25   propagatedBuildInputs = [ utillinux ]; # Dev headers include <uuid/uuid.h>                                                                                                                                                               
 26   buildInputs = [ readline ];
 27   # disallowedReferences = [ autoconf automake libtool gettext libuuid  utillinux ];                        
```

```
  9984     [CC]     kmem.lo
  9985     [CC]     list_sort.lo
  9986     [CC]     logitem.lo
  9987 In file included from ../include/xfs.h:37:0,
  9988                  from crc32.c:37:
  9989 ../include/xfs/linux.h:20:10: fatal error: uuid/uuid.h: No such file or directory
  9990  #include <uuid/uuid.h>
  9991           ^~~~~~~~~~~~~
  9992 compilation terminated.
  9993 make[2]: *** [Makefile:138: crc32selftest] Error 1
  9994 make[2]: *** Waiting for unfinished jobs....
  9995 make[1]: *** [include/buildrules:36: libxfs] Error 2
  9996 make: *** [Makefile:76: default] Error 2
  9997 builder for '/nix/store/nrpsnxzfwqichx3nw0wamng91caqqqd7-xfsprogs-4.14.0-armv7l-unknown-linux-gnueabihf.drv' failed with exit code 2
  9998 error: build of '/nix/store/nrpsnxzfwqichx3nw0wamng91caqqqd7-xfsprogs-4.14.0-armv7l-unknown-linux-gnueabihf.drv' failed
```
```

Reply:
libuuid is afaisee, is just an alias to utillinux. utillinux package contains a bunch binaries like uuidgen etc. I believe build stage generates uuids. Thats why it is in buildInputs. For the headers, utillinux.dev should be in native build inputs. I hope this clears out the confusion.

```
 23   depsBuildBuild = [ buildPackages.stdenv.cc ];
 24   nativeBuildInputs = [ autoconf automake libtool gettext utillinux.dev ] ; #libuuid ];                                                                                                                                                    
 25   propagatedBuildInputs = [ utillinux ]; # Dev headers include <uuid/uuid.h>                                                                                                                                                               
 26   buildInputs = [ readline ];
 27   disallowedReferences = [ autoconf automake libtool gettext libuuid  utillinux utillinux.dev ];                                                                                                                                                       
```
This one leaks references to utillinux.dev. Hmm...

Reply:
```diff
diff --git a/pkgs/tools/filesystems/xfsprogs/default.nix b/pkgs/tools/filesystems/xfsprogs/default.nix
index 569d9b9491e..38d406a5a68 100644
--- a/pkgs/tools/filesystems/xfsprogs/default.nix
+++ b/pkgs/tools/filesystems/xfsprogs/default.nix
@@ -22,7 +22,7 @@ stdenv.mkDerivation rec {
 
   depsBuildBuild = [ buildPackages.stdenv.cc ];
   nativeBuildInputs = [ autoconf automake libtool gettext libuuid ];
-  propagatedBuildInputs = [ utillinux ]; # Dev headers include <uuid/uuid.h>
+  propagatedBuildInputs = [ libuuid ]; # Dev headers include <uuid/uuid.h>
   buildInputs = [ readline ];
 
   enableParallelBuilding = true;
```
should work fine in other words. The question then is just are both `libuuid`s needed, or is the configure script wrong.

Reply:
@Ericson2314 

```
  9993 
  9994 [void@fermat:~/pi]$ nix-build '<nixpkgs>' -A libuuid
  9995 /nix/store/z9lqi01z7r9r3506pi5zf6giw8qp6n38-util-linux-2.32.1-bin
  9997 [void@fermat:~/pi]$ nix-build '<nixpkgs>' -A utillinux
  9998 /nix/store/3p7r50psz9aqbwl87bkb075dddl2n9wx-util-linux-2.32.1-bin
  9994 [void@fermat:~/pi]$ nix-build '<nixpkgs>' -A utillinux.dev
  9995 /nix/store/liv8vrfji3lxagfasybd5ncc2x3dyp0q-util-linux-2.32.1-dev
  9997 [void@fermat:~/pi]$ nix-build '<nixpkgs>' -A libuuid.dev
  9998 /nix/store/hkm3rwdra3l83b1swalphrwv18mgzpsi-util-linux-2.32.1-dev

```

Reply:
It's `utillinuxMinimal`, not `utillinux`.

Reply:
I think we have settld one of our problems now. So lets concentrate on the the other one. Why do we need two libuuids in both nativeBuildInput, buildInput?

Reply:
I'll check tomorrow. This is potentially valid if libuuid is used both at run-time and build-time, but the configure script is wrong in checking runtime if that's the case. Or it's just needed at build time and upstream is wrong doe checking with `CC` not `CC_FOR_BUILD`. So no matter what, upstream is not correct.

Reply:
OK the comment is true. There's no good way to fix the configure script; Autoconf is very incompletely in it's offering of `*_FOR_BUILD` variants. (If only it did it in a generic way...). So I won't bother with any patch for upstream.